pom.xml

```diff
@@ -236,6 +334,132 @@
         </dependencies>
     </dependencyManagement>
 
+    <profiles>
```

Comment:
I guess this only impacts the discovery protocol right now since it will be used by another product. All others should be at Java11 and stay there. As for here, I guess the switch from Java8 to Java1 will simply be a patched version. I don't think we need branches and all for this.

Reply:
Just to make sure, the profile is based on the JDK version currently installed and running Maven? In other words, there's no obvious way to know which version of Java will be used to build when you run Maven, and no easy way either when you pull it down from Nexus? This feels a little too "indeterministic" to me.
Could we just keep Java 8 for now since that's the one we need in DDF, and when we need it for Java 11, we can either change the pom or create a separate branch?
If we keep both, should we at least pick one by default and use a property to build with the other one? And use ranges (e.g., `<jdk>[1.8,1.11)</jdk>`) instead of fixed versions?

Reply:
I missed that last part. I'll change the profile to activate based on the property instead. But, aside from developers' machines, the build is driven from the JenkinsFile which is in the repo and that one specifies JDK8 so it would always be built with JDK8 until it gets changed.